### PR TITLE
Extend Script

### PR DESCRIPTION
**Pull request by fsimonjetz**
Make `script` an object

```
{
    period: Period,
    periodModifier: PeriodModifier,
    uncertain: bool
}
```

### CODE REVIEW

File: ebl/corpus/web/chapter_schemas.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def _split_reconstruction(
     return text, note, parallel_lines
 
 
-def _parse_recontsruction(
+def _parse_reconstruction(
```
Comment (fsimonjetz) [inline ebl/corpus/web/chapter_schemas.py:176]: typo

File: ebl/fragmentarium/application/fragment_schema.py [hunk 2]
```diff
@@ -90,6 +92,22 @@ def make_introduction(self, data, **kwargs):
...
+    period_modifier = ValueEnum(
+        PeriodModifier, required=True, data_key="periodModifier"
+    )
+    uncertain = fields.Boolean(load_default=None)
+
+    @post_load
+    def make_script(self, data, **kwargs) -> Script:
+        return Script(**data)
+
+
```
Comment (khoidt) [inline ebl/fragmentarium/application/fragment_schema.py:110]: Remove `legacy_script` if unused

File: ebl/fragmentarium/application/fragment_updater.py [hunk 3]
```diff
@@ -62,6 +62,17 @@ def update_introduction(
 
         return self._create_result(updated_fragment)
 
+    def update_script(
+        self, number: MuseumNumber, script: Script, user: User
+    ) -> Tuple[Fragment, bool]:
+        fragment = self._repository.query_by_museum_number(number)
+        updated_fragment = fragment.set_script(script)
+
+        self._create_changlelog(user, fragment, updated_fragment)
```
Comment (khoidt) [inline ebl/fragmentarium/application/fragment_updater.py:71]: Typo

File: ebl/fragmentarium/domain/fragment.py [hunk 4]
```diff
@@ -76,6 +77,20 @@ class Introduction:
     parts: Tuple[MarkupPart]
 
 
+@attr.s(auto_attribs=True, frozen=True)
+class Script:
+    period: Period = attr.ib(default=Period.NONE)
+    period_modifier: PeriodModifier = attr.ib(default=PeriodModifier.NONE)
+    uncertain: Optional[bool] = None
```
Comment (khoidt) [inline ebl/fragmentarium/domain/fragment.py:84]: Default `False`, not optional

File: ebl/tests/fragmentarium/test_fragment_matcher_route.py [hunk 5]
```diff
@@ -17,14 +16,14 @@ def test_fragment_matcher_route(client, fragmentarium, user):
         "score": [
             {
                 "museumNumber": "X.326",
-                "script": fragment_2.legacy_script,
+                "script": fragment_2.script.period.abbreviation,
```
Comment (khoidt) [inline ebl/tests/fragmentarium/test_fragment_matcher_route.py:19]: Check

File: ebl/tests/transliteration/test_atf_parser.py [hunk 6]
```diff
@@ -12,6 +15,7 @@
 from ebl.transliteration.domain.language import Language
 from ebl.transliteration.domain.lark_parser import parse_atf_lark
 from ebl.transliteration.domain.line import ControlLine, EmptyLine, Line
+from ebl.transliteration.domain.stage import ABBREVIATIONS as STAGE_ABBREVIATIONS
 from ebl.transliteration.domain.text import Text
 from ebl.transliteration.domain.transliteration_error import TransliterationError
```
Comment (khoidt) [inline ebl/tests/transliteration/test_atf_parser.py:21]: Define `../../transliteration/domain/ebl_atf.lark` and others as variables

**Review by khoidt — APPROVED** [review]

File: ebl/tests/fragmentarium/test_fragment_matcher_route.py (hunk 5 above)
Comment (fsimonjetz) [reply ebl/tests/fragmentarium/test_fragment_matcher_route.py]: It wasn't correct, should have been the whole (dumped) Script object here, too.

File: ebl/fragmentarium/application/fragment_schema.py (hunk 2 above)
Comment (fsimonjetz) [reply ebl/fragmentarium/application/fragment_schema.py:110]: After talking to Enrique, we'll leave it for now and may delete it later on

**Pull request merged**